packages/docusaurus-theme-classic/src/theme/hooks/useKeyboardNavigation.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {useEffect} from 'react';
+
+import ExecutionEnvironment from '@docusaurus/ExecutionEnvironment';
+
+import './styles.css';
+
+// This hook detect keyboard focus indicator to not show outline for mouse users
+// Inspired by https://hackernoon.com/removing-that-ugly-focus-ring-and-keeping-it-too-6c8727fefcd2
+function useKeyboardNavigation(): void {
+  useEffect(() => {
+    if (!ExecutionEnvironment.canUseDOM) {
+      return undefined;
+    }
+
+    const keyboardFocusedClassName = 'navigation-with-keyboard';
+
+    function handleFirstTab(e: KeyboardEvent) {
+      if (e.key === 'Tab') {
+        document.body.classList.add(keyboardFocusedClassName);
+
+        document.removeEventListener('keydown', handleFirstTab);
```

Comment:
For best performance, I add/remove event handlers depending on whether the focus indicator is enabled. Is it worth it? Or we'd better add event handlers once when the `Layout` component is mounted (as it was done before in `Tabs` component), and accordingly remove them only when the component is unmounted (from `useEffect` return).

Reply:
Unless we have a performance issue for tabs, which is unlikely, I'd rather keep the code simpler to read/understand.

But I'm not sure the classList.toggle would work great because it would remove the outline on non-tab key presses?  A user might navigate to inputs with keyboard, and then type in the input, and I think the focus ring should remain in such case?

Reply:
No, the outline will be displayed, you can see this on the page https://deploy-preview-3626--docusaurus-2.netlify.app/classic/docs/styling-layout

![image](https://user-images.githubusercontent.com/4408379/97211158-a9d66000-17cf-11eb-8873-bc1b45e3e5c2.png)